### PR TITLE
ROIs

### PR DESCRIPTION
**Pull request by manics**
~Initial investigation: https://github.com/IDR/idr0062-blin-nuclearsegmentation/blob/e2f19b2e16b090cd35d651638b1290d63f2e6312/experimentA/idr0062-roi-investigation.ipynb~

ROIs have been added to all images apart from:
- `161202_E75.tif`: no segmentation provided
- `Fused_E875.ics`: no segmentation provided
- ~`E875_Zone_3.tif`: Incompatible ZYX dimensions image:[62L, 372L, 350L] labels:(61, 372, 350)~

Note
> 130520_Neural_Rosettes_Manual.tif has 470 non-zero unique values, which matches the 470 feature rows in idr0000-experimentA_features.tsv.
>
> The range of foreground values is [1, 482] with some missing indices, so it is not possible to use the label from 130520_Neural_Rosettes_Manual.tif as a direct index into idr0000-experimentA_features.tsv.
>
> The best assumption is to reindex the label values to be consecutive.

This probably applies to other images

Images with large numbers of ROIs may not work with the current test64 iviewer.

### CODE REVIEW

File: experimentA/upload_features_rois.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    mask_from_binary_image,
+    NoMaskFound,
+)
+from omero_upload import upload_ln_s
+
+DRYRUN = False
+SKIP_IF_EXISTING_ROIS = True
+PROJECT_NAME = 'idr0062-blin-nuclearsegmentation/experimentA'
+OMERO_DATA_DIR = '/data/OMERO'
+NAMESPACE = 'openmicroscopy.org/idr/analysis/original'
```
Comment (sbesson) [inline experimentA/upload_features_rois.py:28]: This reminds me that in https://github.com/IDR/idr-metadata/pull/365/files#diff-6474da02e8703bf37b47be7fa162c096R83, we decided to go for `idr.openmicroscopy.org/study/xxx`. What would be the impact of updating this to `idr.openmicroscop.org/analysis/original` ? In particular how many other file annotations have we created with this namespace?

**Review by sbesson — APPROVED** [review]
One question about the namespace given recent the introduction of IDR study namespace.

Otherwise, I reviewed representative images from each assay and confirmed the labelled images were available as file attachments and generated ROIs can be accessed from the Web viewer

Screen Shot 2019-06-07 at 13 33 18
Screen Shot 2019-06-07 at 13 33 23

File: experimentA/upload_features_rois.py (hunk 1 above)
Comment (manics) [reply experimentA/upload_features_rois.py:28]: `idr.openmicroscop.org/analysis/original` has been used for ROIs/features files which have been attached below the top level study. This includes
- idr0047 https://idr.openmicroscopy.org/webclient/?show=dataset-4451
- idr0052 https://idr.openmicroscopy.org/webclient/?show=image-5514374

Related, idr0001 has a top-level annotation in the namespace `/test/features/annotations`
- https://idr.openmicroscopy.org/webclient/?show=screen-3

Comment (sbesson) [reply experimentA/upload_features_rois.py:28]: Looking at the numbers we are talking about

```
[sbesson@prod67-omeroreadwrite ~]$ /opt/omero/server/OMERO.server/bin/omero hql "select count(f) from FileAnnotation f where f.ns='openmicroscopy.org/idr/analysis/original'"
Using session for public@localhost:4064. Idle timeout: 10 min. Current group: Public
 # | Col1  
---+-------
 0 | 10394 
(1 row)

To see details for object, enter line number.
To move ahead one page, enter 'p'
To re-display list, enter 'r'.
To quit, enter 'q' or just enter.
```

Let us move forward with the same namespace for now to get the study in a publishable form. But opinions welcome regarding the downstream analysis/consumer workflow. /cc @jburel @joshmoore

**Pull request merged**